components/core/src/package/install.rs

```diff
@@ -236,7 +236,14 @@ impl PackageInstall {
 
         // Only insert a PATH entry if the resulting path string is non-empty
         if !rooted_path.is_empty() {
-            env.insert(PATH_KEY.to_string(), rooted_path);
+            if cfg!(windows) {
+                let mut paths: Vec<PathBuf> = env::split_paths(&rooted_path).collect();
```

Comment:
Strictly speaking, you don't have to call `collect` here, since `env::join_paths` can operate on an iterator.

You could get that iterator, `chain` on your Windows system paths, and then pass that to `env::join_paths`.

The `to_string_lossy().to_string()` bit makes me wonder if we might be typing our env map incorrectly, but perhaps that is a question for another day :thinking: 

Granted, this isn't going to be called in a tight loop, or with tons of input data, so extra allocations aren't something to be concerned about, but it could lead to smaller, easier-to-reason-about code.

Reply:
yeah it seems like `BTreeMap<String, OsString>` would be better

Reply:
@mwrock Just to be clear, I think that change (if indeed it is justified) would be something for another PR... pretty sure that'd have a decently sized ripple effect.